batavia/VirtualMachine.js

```diff
@@ -1417,7 +1421,11 @@ batavia.VirtualMachine.prototype.byte_STORE_MAP = function() {
         case batavia.BATAVIA_MAGIC_35a0:
         case batavia.BATAVIA_MAGIC_34:
             var items = this.popn(3);
-            items[0][items[2]] = items[1];
+            if (items[0].__setitem__) {
+                items[0].__setitem__(items[2], items[1]);
+            } else {
+                items[0][items[2]] = items[1];
+            }
```

Comment:
What's the use case for the "javascript" style subscripting? The end goal is to get to "pure" Python objects, so the subscript shouldn't be required, right? Or is there an internal case where subscript syntax is required?